src/renderer/components/MapEditor/ICCAExportDialog.js

```diff
@@ -0,0 +1,494 @@
+import fsWriteStreamAtomic from 'fs-write-stream-atomic'
+import path from 'path'
+import { remote } from 'electron'
+import pump from 'pump'
+import insertCss from 'insert-css'
+import logger from '../../../logger'
+import createZip from '../../create-zip'
+import api from '../../new-api'
+
+const msgs = defineMessages({
+  // Title for ICCA package export dialog
+  title: 'Export an ICCA export package',
+  // Save button
+  save: 'Save',
+  // cancel button
+  cancel: 'Cancel',
+  // Form instructions
+  formInstructions: 'Please answer these questions before exporting.',
+  aboutYouSection: 'About you:',
+  // Question prompts
+  question1Prompt: `
+    Has your community or indigenous people collectively consented to the map
+    and other information being shared with UNEP-WCMC?`,
+  question2Prompt: `
+    Do you want the map and other information to be available for anyone to
+    view and download?`,
+  question3Prompt: `
+    Name of the community, indigenous people, or individual providing the
+    information`,
+  question4Prompt: `
+    If you are an individual, what is your relationship to the indigenous
+    people or community?`,
+  question5Prompt: `
+    Email address, phone number or address where we can contact you. We will
+    contact you to confirm we have received your information and to discuss
+    the information further where needed. We will also contact you periodically
+    to make sure the information is up to date.`,
+  // Yes/no answers
+  answerYes: 'Yes',
+  answerNo: 'No',
+  // Choices for question 4
+  question4Answer1: 'Member of the community/indigenous people',
+  question4Answer2: 'Representative or associate of the community/indigenous people',
+  question4Answer3: 'Representative of a non-governmental organisation',
+  question4Answer4: 'Other',
+  // Placeholder for question 5
+  question5Placeholder: 'Your contact information',
+  // Helper texts
+  requiredAnswer: 'This answer is required.',
+  atLeastOneAnswer: 'Please fill in at least one field.',
+  errorMissingRequired: 'Please fill out all required fields.',
+  // Labels for community name
+  communityOriginalName: 'Original language',
+  communityEnglishName: 'English'
+})
+
+// Insert a tiny lil' shake animation when an error occurs
+// We don't use the makeStyles api here because it can't handle keyframes
+insertCss(`
+  @keyframes shakeX {
+    from,
+    to {
+      transform: translate3d(0, 0, 0);
+    }
+
+    16.67%,
+    50%,
+    83.33% {
+      transform: translate3d(-10px, 0, 0);
+    }
+
+    33.33%,
+    66.67% {
+      transform: translate3d(10px, 0, 0);
+    }
+  }
+
+  .shakeX {
+    animation-name: shakeX;
+    animation-timing-function: ease-in-out;
+    animation-duration: 500ms;
+    animation-delay: 0;
+  }
+`)
+
+const EditDialogContent = ({ onClose }) => {
+  const classes = useStyles()
+  const { formatMessage } = useIntl()
+  const [saving, setSaving] = useState()
+  const [error, setError] = useState(false)
```

Comment:
This is ok for now, but I like to avoid boolean states, and instead have a single state for the component as a string, which can be `idle`, `saving`, `error`. This avoids bugs and impossible states if we ever add things to this component (because what the component should show is essentially a combination of all boolean states right now).

Reply:
I went ahead and made this change in the revision because there's going to be a new dialog state (when there's boundary data to export). Not sure if you'd like this, but there are actually two categories of states in this component: the dialog state, and the form state.

The dialog state will have one of three values:
  - `pending` - initial state, checking geo data
  - `idle` - default state, form is displayed
  - `error` - no boundaries to export, show instructions

And there are three form states:
  - `idle` - default state, user is filling out form
  - `error` - a required form field is missing or invalid
  - `saving` - saving is in progress, disable inputs

I didn't want to combine the state values because the form state is really a subset of when the dialog state is `idle` (that is, displaying the form), and I didn't want to overload the meanings (which would make a "form error" state also means "display the form" in _addition_ to "this form has an error state"). Another way to get around this, if you don't like having two states in one component, is we put the form content into its own component.